connectors/sources/sharepoint_online.py

```diff
@@ -373,13 +385,27 @@ async def _call_api(self, absolute_url):
             headers = {"authorization": f"Bearer {token}"}
             self._logger.debug(f"Calling Sharepoint Endpoint: {absolute_url}")
 
-            async with self._http_session.get(
-                absolute_url,
-                headers=headers,
-            ) as resp:
-                yield resp
+            match method:
+                case HTTPMethod.GET:
+                    async with self._http_session.get(
+                        absolute_url,
+                        headers=headers,
+                    ) as resp:
+                        yield resp
+
+                case HTTPMethod.POST:
+                    async with self._http_session.post(
+                        absolute_url, headers=headers, json=payload
+                    ) as resp:
+                        yield resp
+
+                case _:
+                    self._logger.error(
+                        f"Unsupported http method for SPO client: '{method.value}'."
+                    )
+                    raise InternalServerError("Something unexpected occurred")
```

Comment:
Is it possible to make it without `switch/case`? It starts to look really complex

Reply:
We could use `get_attr`, use the lowercase value of the enum and do some `**kwargs` magic, but don't know, if that really makes it more readable or more obscure in less LOC.

Reply:
It can probably be even methods like:

`async def _get(self, absolute_url)`
`async def _post(self, absolute_urll, payload)`

I think it's more verbose - passing booleans or enums that change the logic of the methods is making it hard.

For example, you can pass `payload` that is non-none and have a method=GET and code will have to handle it - too much complexity. Let's break the method!

Reply:
> We could use get_attr, use the lowercase value of the enum and do some **kwargs magic, but don't know, if that really makes it more readable or more obscure in less LOC.

I think that's already going into some really dark stuff, let's not do `get_attr` :D

Reply:
Splitted it up 👍